triton/nonblocking_stream.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+"""
+triton.nonblocking_stream
+~~~~~~~~
+
+This module provides for a way to write to a Triton Stream without
+blocking to wait for a successful write to the stream.
+It does this by sending the messageover ZeroMQ to tritond.
+
+Adapted from https://github.com/rhettg/BlueOx/blob/master/blueox/network.py
+
+"""
+import logging
+import threading
+import struct
+import atexit
+import datetime
+import decimal
+import time
+
+import zmq
+import msgpack
+
+from . import errors
+from . import config
+
+log = logging.getLogger(__name__)
+
+# We want to limit how many messages we'll hold in memory so if our tritond is
+# unavailable, we don't just run out of memory.  I based this value on rough
+# value of rather large 3k sized messages, and how many we can fit in 10 megs.
+MAX_QUEUED_MESSAGES = 3500
+
+# If we have pending outgoing messages, this is how long we'll wait after
+# being told to exit.
+LINGER_SHUTDOWN_MSECS = 3000
+
+META_STRUCT_FMT = "!B64p64p"
+
+# We're going to include a version byte in our meta struct for future
+# upgrading.  This is very quickly getting into the kind of bit packing I
+# wanted to avoid for a logging infrastructure, but the performance gain is
+# hard to ignore.
+META_STRUCT_VERSION = 0x4
+
+
+def check_meta_version(meta):
+    value, = struct.unpack(">B", meta[0])
+    if value != META_STRUCT_VERSION:
+        raise ValueError(value)
+
+
+def msgpack_encode_default(obj):
```

Comment:
I think I brought this up in the earlier review, but it seems wrong that the non-blocking stream would have it's own serialization configuration right? 

Serializing the data seems like something that should be independent of the stream type.